File: iosApp/iosAppTests/UIComponent/Recents/RecentsStateRobot.swift
import ComposeApp

final class RecentsStateRobot: BaseStateRobot<RecentsUiState, RecentsEvent> {

    override func defaultState() -> RecentsUiState {
        RecentsUiState.Success(
            items: [
                RecentItem(
                    id: "1",
                    name: "Signature Stack Combo",
                    description: "Combo Meal",
                    relativeTime: "2 days ago",
                    imageUrl: nil
                ),
                RecentItem(
                    id: "2",
                    name: "Cookie Swirl Cup",
                    description: "Dessert",
                    relativeTime: "Last week",
                    imageUrl: nil
                ),
            ],
            eventSink: createEventSink()
        )
    }

    func loadingState() -> RecentsUiState {
        RecentsUiState.Loading(eventSink: createEventSink())
    }

    func emptyState() -> RecentsUiState {
        RecentsUiState.Empty(eventSink: createEventSink())
    }
}

File: features/recents/impl/presentation/src/androidDeviceTest/kotlin/com/mockdonalds/app/features/recents/presentation/RecentsStateRobot.kt
package com.mockdonalds.app.features.recents.presentation

import com.mockdonalds.app.core.test.StateRobot
import com.mockdonalds.app.features.recents.api.domain.RecentItem

class RecentsStateRobot : StateRobot<RecentsUiState, RecentsEvent>() {

    override fun defaultState() = RecentsUiState.Success(
        items = listOf(
            RecentItem("1", "Big Mac Combo", "Combo Meal", "2 days ago", null),
            RecentItem("2", "McFlurry Oreo", "Dessert", "Last week", null),
        ),
        eventSink = createEventSink(),
    )
    
    fun emptyState() = RecentsUiState.Empty(
        eventSink = createEventSink(),
    )
}